scripts/modules/task-manager/scope-adjustment.js

```diff
@@ -0,0 +1,828 @@
+/**
+ * scope-adjustment.js
+ * Core logic for dynamic task complexity adjustment (scope-up and scope-down)
+ */
+
+import { z } from 'zod';
+import {
+	log,
+	readJSON,
+	writeJSON,
+	getCurrentTag,
+	readComplexityReport,
+	findTaskInComplexityReport
+} from '../utils.js';
+import {
+	generateObjectService,
+	generateTextService
+} from '../ai-services-unified.js';
+import { findTaskById, taskExists } from '../task-manager.js';
+import analyzeTaskComplexity from './analyze-task-complexity.js';
+import { findComplexityReportPath } from '../../../src/utils/path-utils.js';
+
+/**
+ * Valid strength levels for scope adjustments
+ */
+const VALID_STRENGTHS = ['light', 'regular', 'heavy'];
```

Comment:
should probably move that to src/constants ? but if we're not going to use it anywhere other than here, then we keep it here, though I saw it being used in mcp, cli, etc. so it would be smart to put it in src/constants